vllm/model_executor/weight_utils.py

```diff
@@ -275,6 +276,122 @@ def hf_model_weights_iterator(
             torch.cuda.empty_cache()
 
 
+def kv_cache_scales_loader(
+        filename: str, tp_rank: int, tp_size: int, num_hidden_layers: int,
+        model_type: Optional[str]) -> Iterable[Tuple[int, float]]:
+    """
+    A simple utility to read in KV cache scaling factors that have been
+    previously serialized to disk. Used by the model to populate the appropriate
+    KV cache scaling factors. The serialization should represent a dictionary
+    whose keys are the TP ranks and values are another dictionary mapping layers
+    to their KV cache scaling factors.
+    Keep this function in sync with the output of
+    3rdparty/quantization/extract_scales.pyc
+    """
+
+    # TODO: Once a quantization params format is finalized, pull these Pydantic
+    # schemas out into a separate file that deals solely with quantization
+    # params and their related schemas so that they can be generalized and
+    # shared across various use cases.
+    class KVCacheQuantSchema(BaseModel):
+        dtype: str
+        # Each key is a TP rank. Each value is a dictionary mapping a TP rank's
+        # layer indices to their per-tensor KV cache scaling factor.
+        # TODO: Consider pulling this and its validation methods out into its
+        # own schema class (tricky as its members are variable)
+        scaling_factor: Dict[str, Dict[int, float]]
+
+        @computed_field
+        @property
+        def rank_keyword(self) -> str:
+            # Each TP rank key should be prefixed by a common rank_keyword.
```

Comment:
i didn't realize this is tp dependent. please add this to documentation!

Reply:
Rank keyword has been removed: the extra processing was more trouble than it's worth.